src/components/Counter.js

```diff
@@ -1,39 +1,49 @@
 import React, { Component } from "react";
 import { connect } from 'react-redux';
-import { increment, decrement } from '../actions';
+import { increment, decrement, incrementIfOdd } from '../actions';
 
 class Counter extends Component {
-    incrementIfOdd = () => {
-        // Stretch Problem: Implement an increment function that
-        // only increments if the counter value is odd
+    incrementIfOdd = e => {
+      e.preventDefault();
+      this.props.incrementIfOdd(this.props.count);
```

Comment:
2 things:
- remember that buttons do not have a default behavior to prevent so the preventDefault is not necessary
-instead of making an entirely new action creator, importing it to your component, sending it to your reducer, etc, you can make this function here check a conditional to see if the current count is odd, and if it is then simple invoke the `this.props.increment` action creator you already imported. the rule of thumb here is that if you're making a new action creator but the final effect is has on state is the same as an existing creator, you can invoke that original action creator within your component with some extra logic attached